File: Leetcode/Daily/07-2025/C++/05.cpp
// Problem: 1394. Find Lucky Integer in an Array (Easy)
// Link: https://leetcode.com/problems/find-lucky-integer-in-an-array

class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int, int> umap;
        
        for (int &i: arr) umap[i]++;
        
        int ans = -1;
        for (auto &it: umap) {
            if (it.first == it.second) ans = max(ans, it.first);
        }
        
        return ans;
    }
};

File: Leetcode/Daily/07-2025/Python3/05.py
# Problem: 1394. Find Lucky Integer in an Array (Easy)
# Link: https://leetcode.com/problems/find-lucky-integer-in-an-array

class Solution:
    def findLucky(self, arr: List[int]) -> int:
        umap = {}
        
        for i in arr: 
            umap[i] = umap.get(i, 0) + 1
        
        ans = -1
        for key, value in umap.items():
            if key == value: 
                ans = max(ans, key)
        
        return ans